app/src/main/java/com/vitorpamplona/amethyst/model/LocalCache.kt

```diff
@@ -99,14 +99,18 @@ object LocalCache {
 
     fun checkGetOrCreateChannel(key: String): Channel? {
         return try {
-            val checkHex = Hex.decode(key).toNote() // Checks if this is a valid Hex
+            checkIfValidHex(key)
             getOrCreateChannel(key)
         } catch (e: IllegalArgumentException) {
             Log.e("LocalCache", "Invalid Key to create channel: $key", e)
             null
         }
     }
 
+    private fun checkIfValidHex(key: String) {
+        Hex.decode(key).toNpub()
```

Comment:
I'm not at PC right now, but this method throws? I would put the try catch here and return a Boolean from this function instead.

Reply:
Agree, it can be a `isValidHexNpub` (it throws in both parsing and creating if the key is not valid) with a try catch inside, returning a simple boolean to each check function. 

We will do this in the new refactoring :)

Reply:
Thanks for the feedback! I will do that later :)